src/Surfnet/ServiceProviderDashboard/Application/Console/CleanOldProductionEntitiesCommand.php

```diff
@@ -0,0 +1,104 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Application\Console;
+
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Surfnet\ServiceProviderDashboard\Domain\Repository\EntityRepository;
+use Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Client\QueryClient;
+use Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Exception\QueryServiceProviderException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CleanOldProductionEntitiesCommand extends Command
+{
+    /**
+     * @var EntityRepository
+     */
+    private $entityRepository;
+    /**
+     * @var QueryClient
+     */
+    private $productionManageClient;
+
+    /**
+     * CleanOldProductionEntitiesCommand constructor.
+     * @param EntityRepository $entityRepository
+     * @param QueryClient $manageClient
+     */
+    public function __construct(EntityRepository $entityRepository, QueryClient $productionManageClient)
+    {
+        $this->entityRepository = $entityRepository;
+        $this->productionManageClient = $productionManageClient;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('app:clear-old-entities')
+            ->setDescription('Clears manage production entities not known to Manage.')
+            ->setHelp('This command allows you to clear old published production entities not known in to the production Manage.');
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|null|void
+     * @throws QueryServiceProviderException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $output->write('Start');
+        do {
+            $output->write('.');
+        } while ($this->updateNextBatch());
+        $output->write("\nDone\n");
+    }
+
+    /**
+     * @return bool
+     * @throws QueryServiceProviderException
+     */
+    private function updateNextBatch()
+    {
+        $clearEntities = [];
+        
+        // fetch published production entities
+        $entities = $this->entityRepository->findByState(Entity::STATE_PUBLISHED, Entity::ENVIRONMENT_PRODUCTION);
+        if (empty($entities)) {
+            return false;
+        }
+
+        // check if entities are in manage and if not add to queue
+        foreach ($entities as $entity) {
+            $result = $this->productionManageClient->findByManageId($entity->getManageId());
+            if (is_null($result)) {
+                $clearEntities[] = $entity;
+            }
+        }
+
+        // remove entities in queue
+        foreach ($clearEntities as $entity) {
+            $this->entityRepository->delete($entity);
```

Comment:
Adding log entries for deletion of an entity is a valuable addition.

Reply:
Ahh, miscommunication; I meant logging to a LoggerInterface, not outputting status messages to the console.  This will be sufficient logging for now in my opinion